### PR TITLE
Add PACK, MATH, UNPACK kernels information to METALIUM_GUIDE.md

### PR DESCRIPTION
**Pull request by jaebaek**
Unlike Data Movement kernels, Metalium compile pipeline actually generates three kernels for a Compute kernel, i.e., PACK, MATH, and UNPACK kernels. A Compute kernel developer can see only one kernel i.e., the Compute kernel itself. It is nice to provide the abstraction, but for advanced developers who have to understand the exact details of how Unpack and Pack for tiles work, the information about the three kernels will help them a lot. This commit adds the information to METALIUM_GUIDE.md.

### Ticket
N/A

### Problem description
When I debugged my simple kernel, I met the situation that `pack_tile(..)` does not work. I was not able to debug it by reading known APIs and examples. I wanted to see how the Compute kernel is actually compiled. While investigating on it, I got some useful information, and I believe sharing this with other developers can help them a lot.

### What's changed
I added PACK, MATH, UNPACK kernels information to METALIUM_GUIDE.md.

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by jaebaek** [comment]
I want to link it to one of examples: https://github.com/tenstorrent/tt-metal/tree/main/docs/source/tt-metalium/tt_metal/examples, but not sure what the good example is for this explanation.

**Comment by jaebaek** [comment]
@ayerofieiev-tt  @SeanNijjar Could you please take a look at this PR?

File: METALIUM_GUIDE.md [hunk 1]
```diff
@@ -114,6 +114,152 @@ TODO: Describe that TT wins at scale-out, best compute density at the server an
...
+three Baby RISCVs. tt-metal uses them to conduct unpack, math, and pack. It
+actually generates Unpack, Math, Pack kernels from a single Compute kernel.
+
+For details, let's check what tt-metal generates for the following compute
+kernel:
+```
+namespace NAMESPACE {
+void MAIN {
+  mm_init();
+  acquire_dst(tt::DstMode::Tile);
```
Comment (davorchap) [inline METALIUM_GUIDE.md:129]: this is great, please also use `tile_regs_acquire` , `tile_regs_commit` , `tile_regs_wait` , `tile_regs_release`,   , the new  math->pack sync APIs,

these will further clarify the double buffer hand-shake on the tile_regs registers

File: METALIUM_GUIDE.md [hunk 2]
```diff
@@ -114,6 +114,152 @@ TODO: Describe that TT wins at scale-out, best compute density at the server an
...
+  cb_reserve_back(tt::CB::c_out0, /* number of tiles */ 1);
+  pack_tile(0, tt::CB::c_out0);
+  cb_push_back(tt::CB::c_out0, /* number of tiles */ 1);
+
+  release_dst(tt::DstMode::Tile);
+}
+}  // namespace NAMESPACE
+```
+
+It takes two matrix tiles from `tt::CB::c_in0` and `tt::CB::c_in0` L1 and
```
Comment (davorchap) [inline METALIUM_GUIDE.md:148]: please also add that MATH and pACK synchronize via  tile registers (aka destination registers), and that `tile_regs_` APIs are used to acquire/commit on the math side, and wait/release on the pack side.

File: METALIUM_GUIDE.md [hunk 3]
```diff
@@ -114,6 +114,152 @@ TODO: Describe that TT wins at scale-out, best compute density at the server an
...
+
+On the other hand, the UNPACK kernel has the following
+`cb_wait_front(..)`:
+```
+inline __attribute__((always_inline)) void cb_wait_front(uint32_t cbid, uint32_t ntiles) {
+    ( llk_wait_tiles(cbid, ntiles) );
+}
+```
+
+Another interesting function is `acquire_dst(tt::DstMode mode)`:
```
Comment (davorchap) [inline METALIUM_GUIDE.md:225]: as mentioned above, switch to the new `tile_regs_` APIs.

**Comment by davorchap** [comment]
thank you @jaebaek , this is really good.

I left couple of comments regarding the use of new `tile_regs_` APIs for math->pack sync

**Comment by davorchap** [comment]
> I want to link it to one of examples: https://github.com/tenstorrent/tt-metal/tree/main/docs/source/tt-metalium/tt_metal/examples, but not sure what the good example is for this explanation.

your standalone example is good as a very basic example of what is done on each of unpack, math, pack

**Review by davorchap — APPROVED** [review]

File: METALIUM_GUIDE.md (hunk 1 above)
Comment (jaebaek) [reply METALIUM_GUIDE.md:129]: Ah, I did not know them. Updated. Thanks!

**Comment by davorchap** [comment]
> Thanks! I added some description about `tile_regs_..()` functions. I hope it is clear. Please let me know!

Great!

**Review by davorchap — APPROVED** [review]

**Pull request merged**